### DISCUSSION TITLE
请问开启daemon后，日志写到哪里了？

### DISCUSSION THREAD

**Discussion by jtby** [cloudwu/skynet#1853] [Q&A] [answered]
拉取的最新master分支
通过example/config.userlog启动，只改动了以下两处
```config
logpath = "./log"
daemon = "./skynet.pid"
```
启动后没有看到日志文件，手动新建log目录也没有。
请问该怎么办？

**Comment by JavenLaw** [comment]
第一：配置为 "./log" ，你需要在 skynet 项目下先建立一个名为 log 的文件夹

第二：执行 nc 127.0.0.1  8000  进入 skynet 后台控制 debug console，执行list，显示所有服务的地址

第三：输入：logon 服务地址，例如：logon 8

第四：在 skynet/log/00000008.log 文件，就是这个服务的打印日志（main服务的地址是00000008）

第五：建议 example/main.lua 后面增加：
```lua
           local function my_timeout()
		print("timer=======", skynet.now())

		skynet.timeout(200, my_timeout)
	   end

	   my_timeout()

	  -- skynet.exit()  -- main服务不要退出
```
最后执行 tail -f  00000008.log 就可以查看到 main 服务的日志输出

第六：输入：logoff 服务地址，例如：logoff 8，关闭日志输出

**Reply by jtby** [reply]
谢谢，看到日志文件了。
但是这样操作在实际业务中会有点麻烦，比如游戏中动态创建、销毁的服务无法这样操作。
skynet是否有提供更好用的api? 还是需要我们自己去实现？

**Reply by hanxi** [reply, marked as answer]
一般都是自己实现日志服务的，比如这个。

https://github.com/hanxi/wlua/blob/main/service/log.lua

**Reply by jtby** [reply]
好的，谢谢

**Reply by sniper00** [reply]
> 一般都是自己实现日志服务的，比如这个。
> 
> 
> 
> https://github.com/hanxi/wlua/blob/main/service/log.lua

不建议直接在skynet线程中调用 阻塞的系统API，会影响skynet 调度

**Reply by hanxi** [reply]
哪个api?io.open吗？

**Reply by sniper00** [reply]
> 哪个api?io.open吗？

File write flush

**Comment by hanxi** [comment]
> > 一般都是自己实现日志服务的，比如这个。
> > https://github.com/hanxi/wlua/blob/main/service/log.lua
> 
> 不建议直接在skynet线程中调用 阻塞的系统API，会影响skynet 调度

跟C实现的log服务差不多都是调用的open和write接口，阻塞在log服务应该没问题吧。https://github.com/cloudwu/skynet/blob/master/service-src/service_logger.c

**Comment by hanxi** [comment]
担心性能就用这个吧，好像是万国觉醒的。

https://github.com/lf723/mushroom/blob/master/common/luaclib_src/lua-log.c

https://github.com/lf723/mushroom/blob/master/common/service/syslog.lua
